### ISSUE TITLE
iohyve create doesn't work without a pool name the first time if ISOs were copied

### ISSUE THREAD

**Issue by jmdavis** [pr1ntf/iohyve#210]
I found that the first time that I tried to create a guest via iohyve, `iohyve create` failed if I did not provide a pool name. Digging through the code, it looks like the problem is this bit:

	if [ -z $pool ]; then
		local pool="$(zfs list -H | grep /iohyve/ISO | cut -f1 -d/)"
	fi

Running `zfs list -H | grep /iohyve/ISO | cut -f1 -d/` yielded four lines with `zdata`. I think that the problem was that I had copied over 3 .iso files, and that command was not prepared for that.

I was able to create the guest by specifying the name, and subsequent guests have not been a problem, because

	local pool="${4-$(zfs get -H -s local -o name -t filesystem iohyve:name | head -n1 | cut -f1 -d/)}"

gets used instead, and it works properly once at least one guest is there.

Presumably, that first command should be adjusted so that it just takes the first line of what it currently generates, but my command-line-foo for stuff like this isn't very strong, so I don't know what the correct fix would be.

**Comment by pr1ntf** [comment]
I thought we had this fixed but I may be mistaken. 

Can you tell me what version of `iohyve` you have and how you got it? (ports/pkg or Github)

**Comment by jmdavis** [comment]
I'm use TrueOS. Initially, I used their package, which is currently 0.7.6, and I know that it had the problem. I did update to 0.7.7 using ports, and I _think_ that it still had the problem, because I _think_ that I updated specifically to see if it fixed this problem, but unfortunately, I don't remember now. And I don't know how to reproduce the problem without blowing away all of my virtual machines and everything else from iohyve and starting from scratch (and when I tried to reproduce it by trying to run iohyve inside of a VM in bhyve, it complained that my CPU didn't support bhyve). But looking at the code for 0.7.7, the lines

    if [ -z $pool ]; then
        local pool="$(zfs list -H | grep /iohyve/ISO | cut -f1 -d/)"
    fi

are still there in `ioh-guest` (lines 165 - 167). And if I run that command right now, the pool name is listed 6 times - once for every ISO and once for their parent dataset. I don't know that it's the best solution, but adding an extra grep like so

    $(zfs list -H | grep /iohyve/ISO | grep -v /iohyve/ISO/ | cut -f1 -d/)

does fix it so that that command lists the pool name only once on my system.
